sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -640,11 +640,36 @@ object ScalaJSPluginInternal {
       loadedJSEnv := {
         val log = streams.value.log
         val env = resolvedJSEnv.value
-        val libs =
+        val deps =
           resolvedJSDependencies.value.data ++ scalaJSConfigurationLibs.value
+
+        /* Implement the behavior of commonJSName without having to burn it
+         * inside NodeJSEnv, and hence in the JSEnv API.
+         * Since this matches against NodeJSEnv specifically, it obviously
+         * breaks the OO approach, but oh well ...
+         */
```

Comment:
I'm not entirely sold on the multiple load specs solution yet, true. But I'm not sure how that relates to this PR. A double loading spec would be quite different than `commonJSName`, and in a good direction anyway. I don't think `commonJSName` is the way forward, and neither are `jsDependencies` the way they exist now. `jsDependencies` have serious issues with CommonJSModule support, so much so that they are incompatible. I don't think trying to find an elegant way to deal with `commonJSName` is a priority at this point.